modules/programs/xplr.nix

```
@@ -0,0 +1,68 @@
+{ config, lib, pkgs, ... }:
+with builtins;
+let
+  inherit (lib)
+    types mkIf mkOption mkEnableOption mkPackageOptionMD mdDoc
+    literalExpression;
+
+  cfg = config.programs.xplr;
+  initialConfig = ''
+    version = '${cfg.package.version}'
+  '';
+  # we provide a default version line within the configuration file, which is obtained from the package's attributes
+  # merge the initial configFile, a mapped list of plugins and then the user defined configuration to obtain the final configuration
+  pluginPath = if cfg.plugins != [ ] then
+    (''
+      package.path=
+    '' + (concatStringsSep " ..\n" (map (p: ''"${p}/init.lua;"'') cfg.plugins))
+      + ''
+         ..
+        package.path
+      '')
+  else
+    "\n";
+  configFile = initialConfig + pluginPath + cfg.config;
+in {
+  meta.maintainers = [ maintainers.notashelf ];
+  options.programs.xplr = {
+    enable = mkEnableOption "xplr, terminal UI based file explorer" // {
+      default = true;
+    };
+
+    package = mkPackageOptionMD pkgs "xplr" { };
+
+    plugins = mkOption {
+      type = with types; nullOr (listOf (either package str));
+      default = [ ];
+      defaultText = literalExpression "[]";
+      description = mdDoc ''
+        Plugins to be added to your configuration file. Must be a package, an absolute plugin path, or string
+        to be recognized by xplr. Paths will be relative to $XDG_CONFIG_HOME/xplr/init.lua unless they are absolute.
+      '';
+    };
+
+    # TODO: rename, this is the main configuration
+    config = mkOption {
```

```suggestion
    extraConfig = mkOption {
```
Since it isn't configured in nix but in lua, it should be named `extraConfig`.

---

I think, if we allow string based extraConfig, we shouldn't add the global config version automatically, rather, we should let the users define it. Something like extraConfig.version (required).

This way, if there's a breaking change, xplr will redirect users to https://xplr.dev/en/upgrade-guide with a friendlier error message, and remind them to update the user config as well as the version.

It isn't required if we use `config` (nix), since, then we can go nix's rules on breaking changes.

---

But then again we can use nix's notification system to do the same during system update. Not sure what'd be the best practice here.